---
title: C6310 | Microsoft Docs
ms.custom: ''
ms.date: 2018-06-30
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 15
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 82d8016331a8e849fc08e4b1e1357159d6fd3b2a
ms.sourcegitcommit: 55f7ce2d5d2e458e35c45787f1935b237ee5c9f8
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 08/22/2018
ms.locfileid: "47466317"
---
# <a name="c6310"></a>C6310
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

A versão mais recente deste tópico pode ser encontrada em [C6310](https://docs.microsoft.com/visualstudio/code-quality/c6310).  
  
Aviso C6310: constante ilegal no filtro de exceção pode causar um comportamento inesperado  
  
 Esta mensagem indica que uma constante inválida foi detectada na expressão de filtro de um manipulador de exceção estruturada. As constantes definidas para uso na expressão de filtro de um manipulador de exceção estruturada são:  
  
-   `EXCEPTION_CONTINUE_EXECUTION`  
  
-   `EXCEPTION_CONTINUE_SEARCH`  
  
-   `EXCEPTION_EXECUTE_HANDLER`  
  
 Esses valores são definidos em excpt de arquivo de cabeçalho de tempo de execução.  
  
 Usar uma constante que não está na lista anterior pode causar um comportamento inesperado.  
  
## <a name="example"></a>Exemplo  
 O código a seguir gera este aviso:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Para corrigir este aviso, use o seguinte código:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```



